src/gpb_gen_types.erl

```diff
@@ -444,7 +444,7 @@ msg_to_typestr(M, Opts) ->
 string_to_typestr(true) ->
   "iodata()";
 string_to_typestr(false) ->
-  "iolist()".
+  "binary() | iolist()".
```

Comment:
Actually, the type `iodata()` is the same as `iolist() | binary()` so the two function clauses become identical, which would opens up for some further simplifications.  (For reference, see the table of built-in types a bit down in [section 7.2, _Types and their Syntax_](http://erlang.org/doc/reference_manual/typespec.html#types-and-their-syntax))